[Bug 16280] Make sure the linkhilitecolor returns the correct color

livecodepanos opened the pull request.

livecodepanos committed:
[Bug 16280] Make sure the linkhilitecolor returns the correct color

livecodepanos committed:
[Bug 16280] Added tests for linkhilitecolor

livecodepanos committed:
[Bug 16280] Added bugfix note

livecodesebastien commented on tests/lcs/core/interface/interface.livecodescript:
```diff
@@ -823,4 +823,16 @@ type "a"
+
+   set the linkhilitecolor to "#336633"
+   TestAssert "test3", the linkhilitecolor is "#336633"
+end TestLinkHiliteColors
```
Good idea to add testing - can you add an initial test 
`TestAssert "test1", the linkhilitecolor is empty`

and a final test

```
set the linkhilitecolor to empty
TestAssert "test5", the linkhilitecolor is empty
```

?

livecodepanos replied:
Yes sure. I think the first test should be `TestAssert "test1", the linkhilitecolor is "255,0,0"`, since by default, the (global) linkHiliteColor is set to "255,0,0".

livecodesebastien replied:
The docs states that
`By default, the linkHiliteColor is set to "255,0,0" (bright red). The linkHiliteColor of a newly created stack is set to empty by default.`.
Currently, if you create a new stack, the `linkhilitecolor` is indeed `empty` - but that might not be the case in the tests (needs to be checked I suppose)

livecodepanos committed:
[Bug 16280] Added some more tests

peter-b commented:
@livecodepanos I'm supposed to be writing up some documentation for how to write tests, but for now, could I please ask for the following changes?
1. When you use the `TestAssert` command, the first argument is a description of the test.  For example, `TestAssert "linkhilitecolor hex", the linkhilitecolor is "#336633"`.  This make it faster to figure out what broke by reading the full test logs.
2. Please don't add to these big test files that Ali migrated from the old test suite.  Please create new files for each feature.  For example, I think these tests could go it `tests/lcs/core/interface/linkhilitecolor.livecodescript` (or something like that, ask @runrevmark ...)

livecodepanos committed:
[Bug 16280] Better tests

1. First argument of TestAssert should provide a description of the test
2. Create separate files for each feature being tested

livecodepanos committed:
[Bug 16280] Removed empty new lines

runrevmark commented:
@livecode-vulcan review ok aa4b368

livecode-vulcan commented:
💙 review by @runrevmark ok aa4b368317198d5be3d0f944beaeecdb554d52fe

livecode-vulcan commented:
😎 test success aa4b368317198d5be3d0f944beaeecdb554d52fe
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/863)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/877)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/864)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/867)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/877)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/887)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/880)

livecodesebastien commented on tests/lcs/core/interface/linkHiliteColor.livecodescript:
```diff
@@ -0,0 +1,33 @@
+script "CoreInterfaceLinkHiliteColor"
+/*
+Copyright (C) 2015 Runtime Revolution Ltd.
```
Can you change to `Copyright (C) 2015 LiveCode Ltd.`?

livecodepanos replied:
Nice catch!

livecodepanos committed:
Replaced "Runtime Revolution" with "LiveCode"

runrevmark commented:
@livecode-vulcan review ok e32120e

livecode-vulcan commented:
💙 review by @runrevmark ok e32120e23bebf455f2d5971e80ca64f9e0f46b14

livecode-vulcan commented:
😎 test success e32120e23bebf455f2d5971e80ca64f9e0f46b14
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/869)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/883)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/870)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/873)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/883)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/893)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/886)
